Automattic/jetpack
# [Plugin] Backup: Disconnect on Deactivation if last plugin

This PR adds a deactivation hook to disconnect the Jetpack connection if the Jetpack Backup plugin is the last plugin using the connection.

#### Changes proposed in this Pull Request:

* Adds a deactivation hook
* Implements connection manager's `remove_connection`

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:





- Test plugin solo
* Checkout this branch and build the Backup Plugin
* Activate the Backup Plugin
* Go to Jetpack Backup and setup the connection
* Deactivate the Backup Plugin
* Activate the Backup Plugin
* Go to Jetpack Backup and ensure the connection dialog is present

- Test with Jetpack backup
* Install Jetpack plugin and connect
* Follow `Test plugin solo` instructions, but both times you give the Jetpack Backup plugin page you shouldn't need to connect as a connection should already be established and preserved after disconnection.
